dev/ToastNotifications/NotificationTransientProperties.cpp

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+
+#include "ToastNotification.h"
+
+#include "ToastNotificationUtility.h"
+#include "NotificationTransientProperties.h"
+
+namespace winrt
+{
+    using namespace winrt::Microsoft::Windows::ToastNotifications;
+}
+
+namespace ToastABI
+{
+    using namespace ::ABI::Microsoft::Internal::ToastNotifications;
+}
+
+NotificationTransientProperties::NotificationTransientProperties(winrt::ToastNotification const& toastNotification) noexcept
+{
+    m_suppressPopup = toastNotification.SuppressDisplay();
+
+    if (toastNotification.Priority() == winrt::Microsoft::Windows::ToastNotifications::ToastPriority::High)
+    {
+        m_toastNotificationPriority = ToastABI::ToastNotificationPriority::ToastNotificationPriority_High;
+    }
+}
+
+STDMETHODIMP_(HRESULT __stdcall) NotificationTransientProperties::get_OfflineCacheCount(_Out_ unsigned long long* offlineCacheCount) noexcept
+{
+    *offlineCacheCount = 0;
```

Comment:
Is this property offlineCacheCount always be 0.  Are there any places that we will change its value? If not, why do we need this property?

Reply:
It is useful for push notifications. For local toasts, this property is irrelevant.

While in the SDK layer it's not needed, we still want it in the UDK to translate internal types into the ABI types passed to the LRP.